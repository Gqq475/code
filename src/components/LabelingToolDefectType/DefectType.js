import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { TabList, TabPanel, Tabs, Tab } from '@blueprintjs/core'
import classes from './LabelingToolDefectType.scss'
import {
  MODIFY_DEFECT_MARK_NAME,
  SET_CURRENT_DEFECT_TYPE
} from 'routes/OJS/modules/OJS'

type Props = {
  data: PropTypes.object,
  modifyDefectTypeName: PropTypes.func,
  setCurrentDefectType: PropTypes.func,
  isDialog: PropTypes.bool
};

const DefectTypeItem = (props) => {
  let { index, isSelected, defectTypeId, defectTypeName, handleClick } = props
  return (
    <div
      data-index={index}
      className={+isSelected && classes.active}
      onClick={(e) => handleClick(e.target.dataset.index * 1, defectTypeId, defectTypeName)}
      >
      { defectTypeName }
    </div>
  )
}

DefectTypeItem.propTypes = {
  index: PropTypes.number,
  isSelected: PropTypes.bool,
  defectTypeId: PropTypes.number,
  defectTypeName: PropTypes.string,
  handleClick: PropTypes.func
}

export class DefectType extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
    this.state = {
      General: {},
      Thin: {},
      Photo: {},
      Etch: {},
      Testing: {},
      CF: {},
      selected: ''
    }
  }

  handleClick (index, defectTypeId, defectTypeName) {
    this.setState({
      selected: index
    })
    if (this.props.isDialog) {
      this.props.setCurrentDefectType(defectTypeId, defectTypeName)
    } else {
      this.props.modifyDefectTypeName(defectTypeId, defectTypeName)
    }
  }

  render () {
    return (
      <div>
        <Tabs>
          <TabList className={classes.tabs}>
            <Tab>General</Tab>
            <Tab>Thin Film</Tab>
            <Tab>Photo</Tab>
            <Tab>Etch</Tab>
            <Tab>Testing</Tab>
            <Tab>CF</Tab>
          </TabList>
          <TabPanel className={classes.tabContent}>
            {
              this.props.data &&
              this.props.data.map((item, index) => {
                if (item.departmentCode === 'General') {
                  return (
                    <DefectTypeItem
                      index={index}
                      key={`${item.departmentCode}-${item.id}`}
                      isSelected={this.state.selected === index}
                      defectTypeId={item.id}
                      defectTypeName={item.name}
                      handleClick={this.handleClick} />
                  )
                }
              })
            }
          </TabPanel>
          <TabPanel className={classes.tabContent}>
            {
              this.props.data &&
              this.props.data.map((item, index) => {
                if (item.departmentCode === 'ThinFilm') {
                  return (
                    <DefectTypeItem
                      index={index}
                      key={`${item.departmentCode}-${item.id}`}
                      isSelected={this.state.selected === index}
                      defectTypeId={item.id}
                      defectTypeName={item.name}
                      handleClick={this.handleClick} />
                  )
                }
              })
            }
          </TabPanel>
          <TabPanel className={classes.tabContent}>
            {
              this.props.data &&
              this.props.data.map((item, index) => {
                if (item.departmentCode === 'Photo') {
                  return (
                    <DefectTypeItem
                      index={index}
                      key={`${item.departmentCode}-${item.id}`}
                      isSelected={this.state.selected === index}
                      defectTypeId={item.id}
                      defectTypeName={item.name}
                      handleClick={this.handleClick} />
                  )
                }
              })
            }
          </TabPanel>
          <TabPanel className={classes.tabContent}>
            {
              this.props.data &&
              this.props.data.map((item, index) => {
                if (item.departmentCode === 'Etch') {
                  return (
                    <DefectTypeItem
                      index={index}
                      key={`${item.departmentCode}-${item.id}`}
                      isSelected={this.state.selected === index}
                      defectTypeId={item.id}
                      defectTypeName={item.name}
                      handleClick={this.handleClick} />
                  )
                }
              })
            }
          </TabPanel>
          <TabPanel className={classes.tabContent}>
            {
              this.props.data &&
              this.props.data.map((item, index) => {
                if (item.departmentCode === 'Testing') {
                  return (
                    <DefectTypeItem
                      index={index}
                      key={`${item.departmentCode}-${item.id}`}
                      isSelected={this.state.selected === index}
                      defectTypeId={item.id}
                      defectTypeName={item.name}
                      handleClick={this.handleClick} />
                  )
                }
              })
            }
          </TabPanel>
          <TabPanel className={classes.tabContent}>
            {
              this.props.data &&
              this.props.data.map((item, index) => {
                if (item.departmentCode === 'CF') {
                  return (
                    <DefectTypeItem
                      index={index}
                      key={`${item.departmentCode}-${item.id}`}
                      isSelected={this.state.selected === index}
                      defectTypeId={item.id}
                      defectTypeName={item.name}
                      handleClick={this.handleClick} />
                  )
                }
              })
            }
          </TabPanel>
        </Tabs>
      </div>
    )
  }
}

export default connect(
  (state) => ({}),
  (dispatch) => ({
    setCurrentDefectType: (defectTypeId, defectTypeName) => {
      dispatch({
        type: SET_CURRENT_DEFECT_TYPE,
        data: {
          defectTypeId,
          defectTypeName
        }
      })
    },
    modifyDefectTypeName: (defectTypeId, defectTypeName) => {
      dispatch({
        type: MODIFY_DEFECT_MARK_NAME,
        data: {
          defectTypeId,
          defectTypeName
        }
      })
    }
  })
)(DefectType)
