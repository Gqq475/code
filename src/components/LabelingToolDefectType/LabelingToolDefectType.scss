.LabelingToolDefectType-container {
  margin: 20px 0 ;

  .tabs {
    display: block;

    li {
      border-radius: 2px 2px 0 0;
      border: solid 1px #0087DC;
      padding: 0 10px;
      margin: -1px 0 0 -1px;
      display: inline-block;
    }
  }
  .twoTab {
    margin: 0px;
  }
  .tabContent {
    border: solid 1px #0087DC;
    margin: 0px;
  }
  .tabContent div {
    font-size: 12px;
    cursor: pointer;
    padding: 10px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .tabContent div:hover {
    background: #0087DC;
    color: #fff;
  }
  .active {
    background:#b0e0e6;
    color: #666;
  }
}
