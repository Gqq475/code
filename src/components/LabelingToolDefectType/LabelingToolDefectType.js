import React from 'react'
import classes from './LabelingToolDefectType.scss'
import { TabList, TabPanel, Tabs, Tab } from '@blueprintjs/core'
import DefectType from './DefectType'
import { isEmpty } from 'lodash'

type Props = {
  data: Object,
  getDefectTypeList: Function,
  isDialog: Boolean
};
export class LabelingToolDefectType extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.state = {
      TFT: {},
      CF: {},
      Cell: {},
      Module: {}
    }
  }

  componentWillMount () {
    this.props.getDefectTypeList()
  }

  render () {
    return (
      <div className={classes['LabelingToolDefectType-container']}>
        {
          !isEmpty(this.props.data.defectTypeList) &&
            <Tabs key={'vertical'}>
              <TabList className={classes.tabs}>
                <Tab>TFT</Tab>
                <Tab>CF</Tab>
                <Tab>Cell</Tab>
                <Tab>Module</Tab>
              </TabList>
              <TabPanel className={classes.twoTab}>
                <DefectType
                  isDialog={this.props.isDialog}
                  data={this.props.data.defectTypeList.TFT} />
              </TabPanel>
              <TabPanel className={classes.twoTab}>
                <DefectType
                  isDialog={this.props.isDialog}
                  data={this.props.data.defectTypeList.CF} />
              </TabPanel>
              <TabPanel className={classes.twoTab}>
                <DefectType
                  isDialog={this.props.isDialog}
                  data={this.props.data.defectTypeList.Cell} />
              </TabPanel>
              <TabPanel className={classes.twoTab}>
                <DefectType
                  isDialog={this.props.isDialog}
                  data={this.props.data.defectTypeList.Module} />
              </TabPanel>
            </Tabs>
        }
      </div>
    )
  }
}

export default LabelingToolDefectType
