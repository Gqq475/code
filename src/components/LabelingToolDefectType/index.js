import LabelingToolDefectType from './LabelingToolDefectType'
export default LabelingToolDefectType
