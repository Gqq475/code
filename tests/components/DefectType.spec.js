import React from 'react'
import {DefectType} from 'components/LabelingToolDefectType/DefectType'

describe('(Component) DefectType', () => {
  it('render normally if data is not exist and departmentCode is General', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = [
      {
        departmentCode: 'General',
        factoryCode: 'CF',
        id: 2,
        isDelete: false,
        name: 'TestDefect2'
      }
    ]
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if data is not exist and departmentCode is Etch', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = [
      {
        departmentCode: 'Etch',
        factoryCode: 'CF',
        id: 2,
        isDelete: false,
        name: 'TestDefect2'
      }
    ]
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if data is not exist and departmentCode is Photo', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = [
      {
        departmentCode: 'Photo',
        factoryCode: 'CF',
        id: 2,
        isDelete: false,
        name: 'TestDefect2'
      }
    ]
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if data is not exist and departmentCode is ThinFilm', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = [
      {
        departmentCode: 'ThinFilm',
        factoryCode: 'CF',
        id: 2,
        isDelete: false,
        name: 'TestDefect2'
      }
    ]
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if data is not exist and departmentCode is CF', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = [
      {
        departmentCode: 'CF',
        factoryCode: 'CF',
        id: 2,
        isDelete: false,
        name: 'TestDefect2'
      }
    ]
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if data is not exist and departmentCode is Testing', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = [
      {
        departmentCode: 'Testing',
        factoryCode: 'CF',
        id: 2,
        isDelete: false,
        name: 'TestDefect2'
      }
    ]
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if data is exist', () => {
    const getId = jest.fn()
    const modifyDefectName = jest.fn()
    const data = []
    const wrapper = shallow(<DefectType data={data} getId={getId} modifyDefectName={modifyDefectName} />)
    expect(wrapper).toMatchSnapshot()
  })
})
