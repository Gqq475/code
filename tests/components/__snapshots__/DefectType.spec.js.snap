// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DefectType render normally if data is exist 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
  </Tabs>
</div>
`;

exports[`(Component) DefectType render normally if data is not exist and departmentCode is CF 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    >
      <DefectTypeItem
        defectTypeId={2}
        defectTypeName="TestDefect2"
        handleClick={[Function]}
        index={0}
        isSelected={false}
      />
    </TabPanel>
  </Tabs>
</div>
`;

exports[`(Component) DefectType render normally if data is not exist and departmentCode is Etch 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    >
      <DefectTypeItem
        defectTypeId={2}
        defectTypeName="TestDefect2"
        handleClick={[Function]}
        index={0}
        isSelected={false}
      />
    </TabPanel>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
  </Tabs>
</div>
`;

exports[`(Component) DefectType render normally if data is not exist and departmentCode is General 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    >
      <DefectTypeItem
        defectTypeId={2}
        defectTypeName="TestDefect2"
        handleClick={[Function]}
        index={0}
        isSelected={false}
      />
    </TabPanel>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
  </Tabs>
</div>
`;

exports[`(Component) DefectType render normally if data is not exist and departmentCode is Photo 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    >
      <DefectTypeItem
        defectTypeId={2}
        defectTypeName="TestDefect2"
        handleClick={[Function]}
        index={0}
        isSelected={false}
      />
    </TabPanel>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
  </Tabs>
</div>
`;

exports[`(Component) DefectType render normally if data is not exist and departmentCode is Testing 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    >
      <DefectTypeItem
        defectTypeId={2}
        defectTypeName="TestDefect2"
        handleClick={[Function]}
        index={0}
        isSelected={false}
      />
    </TabPanel>
    <TabPanel
      className="tabContent"
    />
  </Tabs>
</div>
`;

exports[`(Component) DefectType render normally if data is not exist and departmentCode is ThinFilm 1`] = `
<div>
  <Tabs
    initialSelectedTabIndex={0}
  >
    <TabList
      className="tabs"
    >
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        General
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Thin Film
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Photo
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Etch
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        Testing
      </Tab>
      <Tab
        isDisabled={false}
        isSelected={false}
      >
        CF
      </Tab>
    </TabList>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    >
      <DefectTypeItem
        defectTypeId={2}
        defectTypeName="TestDefect2"
        handleClick={[Function]}
        index={0}
        isSelected={false}
      />
    </TabPanel>
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
    <TabPanel
      className="tabContent"
    />
  </Tabs>
</div>
`;
